src/jit/codegenxarch.cpp

```diff
@@ -185,16 +185,15 @@ void                CodeGen::genEmitGSCookieCheck(bool pushReg)
         // Handle multi-reg return type values
         if (compiler->compMethodReturnsMultiRegRetType())
         {
-#ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
             ReturnTypeDesc retTypeDesc;
             retTypeDesc.Initialize(compiler, compiler->info.compMethodInfo->args.retTypeClass);
+            unsigned regCount = retTypeDesc.GetReturnRegCount();
+            assert(regCount == MAX_RET_REG_COUNT);
```

Comment:
For HFA returns on Arm32/Arm64 we can have 2, 3 or 4 registers being returned.
So this assert would need to be changed